### PR TITLE
Updating FreeBSD to 12.1 target

### PR DESCRIPTION
**Pull request by jaredwolff**
Restoring FreeBSD to its former glory! A few strategic changes to update the toolchain for compiling on FreeBSD 12.1.

### CODE REVIEW

**Comment by jaredwolff** [comment]
Hey @Dylan-DPC,

Not sure what's up with the solaris builds. Looks like that's happening across other PRs as well. Let me know if you need anything else from me on my end to make this PR happen.

Thanks!

**Comment by Dylan-DPC** [comment]
Thanks. I'm not sure will have to check. Sadly this PR will have to wait for that issue to be fixed just to ensure the build passes and doesn't regress further. I'll let you know if i have any updates

**Comment by jaredwolff** [comment]
Thanks @Dylan-DPC. I totally understand on that end. Happy to rebase and re-test the PR once they are taken care of.

File: docker/freebsd-extras.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    # Adding sqlite3
+    curl --retry 3 -sSfL "https://pkg.freebsd.org/FreeBSD:12:${bsd_arch}/quarterly/All/sqlite3-${sqlite_ver}.txz" -O
+    tar -C "${td}/sqlite" -xJf sqlite3-${sqlite_ver}.txz /usr/local/lib
+
+    # Copy the linked library
+    local destdir="/usr/local/${target}"
+    cp "${td}/openssl/usr/local/lib"/lib{crypto,ssl}.a "${destdir}/lib"
+    cp "${td}/openssl/usr/local/lib"/lib{crypto,ssl}.so.11 "${destdir}/lib"
+    cp "${td}/openssl/usr/local/lib"/lib{crypto,ssl}.so "${destdir}/lib"
+    cp -r "${td}/openssl/usr/local/include" "${destdir}"
```
Comment (reitermarkus) [inline docker/freebsd-extras.sh:43]: Move this to the top.

File: docker/freebsd.sh [hunk 2]
```diff
@@ -6,12 +6,12 @@ set -euo pipefail
 main() {
     local arch="${1}"
 
-    local binutils=2.25.1 \
-          gcc=5.3.0 \
-          target="${arch}-unknown-freebsd10"
+    local base_release=12.1
```
Comment (reitermarkus) [inline docker/freebsd.sh:9]: Missing `\` here.

**Review by reitermarkus — CHANGES_REQUESTED** [review]

File: docker/freebsd-extras.sh (hunk 1 above)
Comment (jaredwolff) [reply docker/freebsd-extras.sh:43]: Do you mean `freebsd.sh`? I was debating to have them separate or together since the extras are not 100% required.

Comment (jaredwolff) [reply docker/freebsd-extras.sh:43]: Do you mean freebsd.sh? I was debating to have them separate or together since the extras are not 100% required.

Comment (reitermarkus) [reply docker/freebsd-extras.sh:43]: I just mean this line specifically. It's kind of out of place in the middle here.

**Comment by jaredwolff** [comment]
@reitermarkus made those changes. Let me know if you see anything else that needs to be reworked.

**Review by reitermarkus — APPROVED** [review]

**Pull request merged**